app/main.py

```diff
@@ -2,5 +2,19 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def inner(*args) -> str:
+        if func.__name__ not in cached_results:
+            print("Calculating new result")
+            cached_results[func.__name__] = {args: func(*args)}
+        else:
+            if args in cached_results[func.__name__]:
+                print("Getting from cache")
+            else:
+                print("Calculating new result")
+                cached_results[func.__name__][args] = func(*args)
+
+        return cached_results[func.__name__][args]
+
+    return inner
```

Comment:
This is a code style issue. The inner function `inner` is not descriptive. Consider renaming it to something more meaningful, like `cached_func`.